internal/promotion/runner/builtin/schemas/git-merge-pr-config.json

```diff
@@ -0,0 +1,47 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "title": "GitMergePRConfig",
+    "type": "object",
+    "additionalProperties": false,
+    "required": ["repoURL", "prNumber"],
+    "properties": {
+      "repoURL": {
+        "type": "string",
+        "description": "The URL of the remote Git repository containing the pull request.",
+        "minLength": 1,
+        "format": "uri"
+      },
+      "prNumber": {
+        "type": "integer",
+        "description": "The number of the pull request to merge.",
+        "minimum": 1
+      },
+      "provider": {
+        "type": "string",
+        "description": "The name of the Git provider to use. Currently 'azure', 'bitbucket', 'gitea', 'github', and 'gitlab' are supported. Kargo will try to infer the provider if it is not explicitly specified.",
+        "enum": ["azure", "bitbucket", "gitea", "github", "gitlab"]
+      },
+      "mergeMethod": {
+        "type": "string",
+        "description": "The merge strategy to use. If not specified, the provider's default will be used.",
+        "enum": ["merge", "squash", "rebase"]
+      },
+      "insecureSkipTLSVerify": {
+        "type": "boolean",
+        "description": "Skip TLS verification when interacting with the Git provider. Default is false."
+      },
+      "deleteBranchAfterMerge": {
+        "type": "boolean",
+        "description": "If true, deletes the source branch after merging the PR. Default is false."
+      },
+      "waitForChecks": {
+        "type": "boolean",
+        "description": "If true, waits for all required status checks to pass before merging. Default is true."
+      },
+      "maxWaitSeconds": {
+        "type": "integer",
+        "description": "Maximum number of seconds to wait for checks to pass. Default is 600 (10 minutes).",
+        "minimum": 1
+      }
```

Comment:
This looks like it would be problematic.

Including this in step configuration seems to imply the step itself is going to wait _internally_ and we ought not do that. "Idling" steps unnecessarily consumes a portion of the controller's finite concurrency. With enough of that going on, the system can become live-locked.

This should be configured using [retries](https://docs.kargo.io/user-guide/reference-docs/promotion-templates#step-retries) instead. (The step can specify its own default timeout that overrides the system-wide default.) Using this approach, Promotions that are just _waiting_ for something external to happen do not consume capacity.